ConcernsCaseWork/ConcernsCaseWork.CypressTests/cypress/pages/caseActions/viewDecisionPage.ts

```diff
@@ -1,18 +1,183 @@
+		cy.task(
+			"log", `With Date ESFA Year ${dateESFAYear}`
+		);
+
+		cy.getById("dtr-year-request-received").clear().type(dateESFAYear);
+
+		return this;
+	}
+    public withTypeOfDecisionID(typeOfDecisionID: string): this {
+		cy.task("log", `With Notice To Improve ${typeOfDecisionID}`);
+
+		cy.getById(typeOfDecisionID).click();
+
+		return this;
+	}
+
+	public withTotalAmountRequested(totalAmountRequested: string): this {
+		cy.task("log", `With total Amount Requested ${totalAmountRequested}`);
+
+		cy.getById("total-amount-request").clear().type(totalAmountRequested);
+
+		return this;
+	}
+
+	
+	public withSupportingNotes(supportingNotes: string): this {
+		cy.task("log", `With Supporting Notes ${supportingNotes}`);
+
+		cy.getById("case-decision-notes").clear().type(supportingNotes);
+
+		return this;
+	}
+
+
+	public hasCrmEnquiry(crmNumber: string): this {
+		cy.task("log", `Has CRM enquiry ${crmNumber}`);
+
+		cy.getByTestId("crm-enquiry-text").should("contain.text", crmNumber);
+
+		return this;
+	}
+
+	public hasRetrospectiveRequest(retrospectiveRequest: string): this {
+		cy.task("log", `Has retrospective request ${retrospectiveRequest}`);
+
+		cy.getByTestId("retrospective-request-text").should(
+			"contain.text",
+			retrospectiveRequest
+		);
+
+		return this;
+	}
+	public hasSubmissionRequired(submissionRequired: string): this {
+		cy.task("log", `Has Submission Required ${submissionRequired}`);
+
+		cy.getByTestId("submission-required-text").should(
+			"contain.text",
+			submissionRequired
+		);
+
+		return this;
+	}
+
+	public hasSubmissionLink(submissionLink: string): this {
+		cy.task("log", `Has Submission link ${submissionLink}`);
+
+		cy.getByTestId("submission-link-text").should(
+			"contain.text",
+			submissionLink
+		);
+
+		return this;
+	}
+	public hasDateESFAReceivedRequest(dateESFAReceivedRequest: string): this {
+		cy.task("log", `Has Date ESFA Received Request ${dateESFAReceivedRequest}`);
+
+		cy.getByTestId("date-esfa-received-text").should(
+			"contain.text",
+			dateESFAReceivedRequest
+		);
+
+		return this;
+	}
+	public hasTotalAmountRequested(totalAmountRequested: string): this {
+		cy.task("log", `Has total Amount Requested ${totalAmountRequested}`);
+
+		cy.getByTestId("amount-requested-text").should(
+			"contain.text",
+			totalAmountRequested
+		);
+
+		return this;
+	}
+	public hasTypeOfDecision(typeOfDecision: string): this {
+		cy.task("log", `Has type of decision  ${typeOfDecision}`);
+
+
+		cy.getByTestId("decision-type-text").should(
+			"contain.text",
+			typeOfDecision
+		);
+
+		return this;
+	}
+	public hasSupportingNotes(supportingNotes: string): this {
+		cy.task("log", `Has Supporting Notes ${supportingNotes}`);
+
+		cy.getByTestId("supporting-notes-text").should(
+			"contain.text",
+			supportingNotes
+		);
+
+		return this;
+	}
+	public hasActionEdit(actionEdit: string): this {
+		cy.task("log", `Click on Edit ${actionEdit}`);
+
+		cy.getByTestId("edit-decision-text").click();
```

Comment:
For clarity I would do the click as a separate action

hasActionEdit, is a check to see if the button rendered

There should be another user action to click it, e.g. EditDecision()